hub/core/chunk_engine.py

```diff
@@ -352,17 +444,22 @@ def update(self, index: Index, samples: Union[Sequence[SampleValue], SampleValue
 
         chunks_nbytes_after_updates = []
         global_sample_indices = tuple(index.values[0].indices(self.num_samples))
-        for i, (buffer, shape) in enumerate(serialized_input_samples):
+        buffer, nbytes, shapes = serialized_input_samples
+        for i, (nb, shape) in enumerate(zip(nbytes, shapes)):
             global_sample_index = global_sample_indices[i]  # TODO!
-
             chunk = self.get_chunk_for_sample(global_sample_index, enc)
-
             local_sample_index = enc.translate_index_relative_to_chunks(
                 global_sample_index
             )
-
             tensor_meta.update_shape_interval(shape)
-            chunk.update_sample(local_sample_index, buffer, shape)
+            chunk.update_sample(
+                local_sample_index,
+                buffer[:nb],  # type: ignore
+                shape,
+                chunk_compression=self.tensor_meta.chunk_compression,
```

Comment:
should we store chunk compression inside the chunk headers?

Reply:
I don''t think we should duplicate info per chunk like that.